dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/GenericFilter.java

```diff
@@ -90,7 +94,9 @@ public Result invoke(Invoker<?> invoker, Invocation inv) throws RpcException {
                     generic = RpcContext.getContext().getAttachment(GENERIC_KEY);
                 }
 
-                if (StringUtils.isEmpty(generic)
+                if (ProtocolUtils.isGsonGenericSerialization(generic)) {
```

Comment:
Code clean: This case should after origin first `if` judgement.

Reply:
agree with horizonzy

Reply:
get